File: momhelp-frontend/momhelp-frontend/src/services/shoppingListService.js
import api from './api';

const shoppingListService = {
  addItem: (item) => {
    return api.post('/shopping/add', item);
  },

  updateItem: (id, item) => {
    return api.put(`/shopping/${id}`, item);
  },

  markAsPurchased: (id) => {
    return api.patch(`/shopping/${id}/purchase`);
  },

  markAsUnpurchased: (id) => {
    return api.patch(`/shopping/${id}/unpurchase`);
  },

  getAllItems: (userId) => {
    return api.get(`/shopping/user/${userId}`);
  },

  getPendingItems: (userId) => {
    return api.get(`/shopping/user/${userId}/pending`);
  },

  getPurchasedItems: (userId) => {
    return api.get(`/shopping/user/${userId}/purchased`);
  },

  getItemsByCategory: (userId, category) => {
    return api.get(`/shopping/user/${userId}/category/${category}`);
  },

  getItemsByPriority: (userId, priority) => {
    return api.get(`/shopping/user/${userId}/priority/${priority}`);
  },

  deleteItem: (id) => {
    return api.delete(`/shopping/${id}`);
  },

  clearPurchasedItems: (userId) => {
    return api.delete(`/shopping/user/${userId}/purchased`);
  },

  clearAllItems: (userId) => {
    return api.delete(`/shopping/user/${userId}/all`);
  }
};

export default shoppingListService;

File: momhelp/src/main/java/com/momhelp/service/ShoppingListService.java
package com.momhelp.service;

import com.momhelp.dto.ShoppingListRequestDTO;
import com.momhelp.dto.ShoppingListResponseDTO;

import java.util.List;

public interface ShoppingListService {

	ShoppingListResponseDTO addItem(ShoppingListRequestDTO requestDTO);

	ShoppingListResponseDTO updateItem(Long id, ShoppingListRequestDTO requestDTO);

	ShoppingListResponseDTO markAsPurchased(Long id);

	ShoppingListResponseDTO markAsUnpurchased(Long id);

	List<ShoppingListResponseDTO> getAllItems(Long userId);

	List<ShoppingListResponseDTO> getPendingItems(Long userId);

	List<ShoppingListResponseDTO> getPurchasedItems(Long userId);

	List<ShoppingListResponseDTO> getItemsByCategory(Long userId, String category);

	List<ShoppingListResponseDTO> getItemsByPriority(Long userId, String priority);

	void deleteItem(Long id);

	void clearPurchasedItems(Long userId);

	void clearAllItems(Long userId);
}